### DISCUSSION TITLE
A stupid question concerning bsCookies

### DISCUSSION THREAD

**Discussion by kreincke** [bootscore/bootscore#559] [Q&A]
If I understood bsCookies correctly, then the cookie dialog itself saves a cookie concerning the values 

- {"level":["necessary"],"revision":0,"data":null,"rfc_cookie":false} OR
- {"level":["necessary","analytics"],"revision":0,"data":null,"rfc_cookie":false} OR
- {"level":["necessary","advertising"],"revision":0,"data":null,"rfc_cookie":false} OR
- {"level":["necessary","analytics","advertising"],"revision":0,"data":null,"rfc_cookie":false}

IN accordance with that classification, we can decide to which group our cookies belong and can add them to the respective groups in the bsCookie script.

Now my - perhaps - a bit stupid question: Is it correct that we, the side creators themselves have to implement the semantics of this cookie handling? Did I understand it correctly, that there isn't any automatism that prevents us or any of our plugins from writing - for example - any analytic cookie if our user has decided 'only necessary cookies'? Is it correct, if I said that having this cookie-dialog activated is only the one half of an adequate cookie handling?

Many thanks for your answers!

**Comment by crftwrk** [comment]
> Is it correct that we, the side creators themselves have to implement the semantics of this cookie handling?

Yes, of course. Plugin does not manage or block cookies by itself, it allows you manage additional cookies like Google Analytics etc.

Note that this plugin uses the modified cookie consent script by Orest Bida https://github.com/orestbida/cookieconsent. There are no modifications in cookie handling, only changed to Bootstrap classes and removed some styling options. If you have questions about the cookie functionality, it's better to ask there.

See here as well https://github.com/orgs/bootscore/discussions/387.
